activesupport/lib/active_support/message_encryptor.rb

```diff
@@ -22,6 +22,31 @@ module ActiveSupport
   #   crypt = ActiveSupport::MessageEncryptor.new(key)                            # => #<ActiveSupport::MessageEncryptor ...>
   #   encrypted_data = crypt.encrypt_and_sign('my secret data')                   # => "NlFBTTMwOUV5UlA1QlNEN2xkY2d6eThYWWh..."
   #   crypt.decrypt_and_verify(encrypted_data)                                    # => "my secret data"
+  #
+  # MessageEncryptor supports the following metadata that's associated with a message:
+  #
+  # * <tt>:purpose</tt>    - The purpose for which the message is intended to be used.
+  # * <tt>:expires_at</tt> - The expiration date of the message.
+  # * <tt>:expires_in</tt> - The relative expiration date. Note that, +:expires_at+
+  # takes precedence over +:expires_in+. Also, if both, +:expires_at+ and +:expires_in+
+  # are not set,then expiration is turned off.
+  #
+  # For example,
+  #
+  #    # Generates an encrypted message purposed for shipping and expires in 1 year.
+  #    address = "#23 main street"
+  #    encrypted_data = crypt.encrypt_and_sign(address, purpose: "shipping", expires_at: 1.year.from_now) # => "B/yFvIcSzfaVQcjQYU+pGq..."
+  #
+  # However, trying to use a message for a different purpose or trying to use an expired
+  # message would fail and return +nil+.
+  #
+  #    crypt.decrypt_and_verify(encrypted_data, purpose: "login")                 # => nil
+  #    crypt.decrypt_and_verify(encrypted_data, purpose: :shipping)               # => "#23 main street"
+  #
+  # You could use the +:expires_in+ option, to set the expiration date relatively.
+  #
+  #    # Generates an encrypted message that expires in 2 months.
+  #    encrypted_data = crypt.encrypt_and_sign(address, expires_in: 2.months)     # => "Dwx8ht2FWhtQb.."
```

Comment:
Personally, I'm not too much of a fan of the raw options list and prefer to read things more as a "story" kind of.

How does this sound?

```
# === Confining messages to a specific purpose
#
# By default any message can be used throughout your app. But they can also be
# confined to a specific +:purpose+.
# 
#   token = crypt.encrypt_and_sign("this is the chair", purpose: :login)
#
# Then that same purpose must be passed when verifying to get the data back out:
#
#   crypt.decrypt_and_verify(token, purpose: :login)    # => "this is the chair"
#   crypt.decrypt_and_verify(token, purpose: :shipping) # => nil
#   crypt.decrypt_and_verify(token)                     # => nil
#
# Likewise, if a message has no purpose it won't be returned when verifying with
# a specific purpose.
# 
#   token = crypt.encrypt_and_sign("the conversation is lively")
#   crypt.decrypt_and_verify(token, purpose: :scare_tactics) # => nil
#   crypt.decrypt_and_verify(token)                          # => "the conversation is lively"
#
# === Making messages expire
#
# By default messages last forever and verifying one years from now will still
# return the original value. But messages can be set to expire at a given
# time with +:expires_in+ or +:expires_at+.
#
#   crypt.encrypt_and_sign(parcel, expires_in: 1.month)
#   crypt.encrypt_and_sign(doowad, expires_at: Time.now.end_of_year)
#
# Then the messages can be verified and returned upto the expire time.
# Thereafter, verifying returns +nil+.
```

(Note: we don't indent code by 4 spaces, only 3 after `#`)